Fix missing include in externs.h

ismaell opened the pull request:
The time.h header is required to make the timespec struct available for
the prototypes.

Fixes: https://github.com/cisco/ChezScheme/issues/281

burgerrg reviewed (changes requested):
Thank you for this fix! Now that externs.h includes time.h, please remove the include of time.h from the C files that already include externs.h. Then update LOG.

ismaell commented:
@burgerrg just to clarify the update: I've addressed both things (it required to make the inclusion unconditional) and squashed everything into a single commit.

burgerrg commented:
Thank you! Do we need the `#include <sys/time.h>` lines in scheme.c and stats.c?

burgerrg commented:
Thanks again for this fix!

ismaell commented:
> On macOS, the sys/time.h header is needed for stats.c.

Neither header requires the other's definitions, so I think that's universal.
